docs/proposals/scheduling/policy-preemption/README.md

```diff
@@ -0,0 +1,256 @@
+
+## Design Details
+
+### API change
+
+#### PropagationPolicy API change
+
+This proposal proposes a new field `Preemption` for determining the behavior for preempting.
+By default, preemption is disabled.
+
+```go
+type PropagationSpec struct {
+    // +kubebuilder:default=0
+    Priority *int `json:"priority,omitempty"`
+
+    // Preemption declares the behaviors for preempting.
+    // Valid options are Always and Never.
+    //
+    // +kubebuilder:default="Never"
+    // +kubebuilder:validation:Enum=Always;Never
+    // +optional
+    Preemption PreemptionBehavior `json:"preemption,omitempty"`
+}
+```
+
+```go
+// PreemptionBehavior describes whether and how to preempt resources that are
+// claimed by lower-priority PropagationPolicy(ClusterPropagationPolicy).
+// +enum
+type PreemptionBehavior string
+
+const (
+    // PreemptAlways means that preemption is allowed.
+    //
+    // If it is applied to a PropagationPolicy, it can preempt any resource as
+    // per Priority, regardless of whether it has been claimed by a PropagationPolicy
+    // or a ClusterPropagationPolicy, as long as it can match the rules defined
+    // in ResourceSelector. In addition, if a resource has already been claimed by a 
+    // ClusterPropagationPolicy, the PropagationPolicy can still preempt it
+    // without considering Priority.
+    //
+    // If it is applied to a ClusterPropagationPolicy, it can only preempt from
+    // ClusterPropagationPolicy, and from PropagationPolicy is not allowed.
+    PreemptAlways PreemptionBehavior = "Always"
+
+    // PreemptNever means that a PropagationPolicy(ClusterPropagationPolicy) never
+    // preempts resources.
+    PreemptNever PreemptionBehavior = "Never"
+)
+```
+
+For example, assume that policy `nginx-propagation` has bound workload `nginx`:
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx-propagation
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  priority: 2 
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member1
+        - member2
+```
+
+```yaml
+apiVersion: work.karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  labels:
+    propagationpolicy.karmada.io/name: nginx-propagation
+    propagationpolicy.karmada.io/namespace: default
+  name: nginx-deployment
+  namespace: default 
+```
+
+Assume that there is a high-priority policy which allows preepmtion:
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx-propagation-preemption
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  priority: 3
+  preemption: Always
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member3
+```
+
+`nginx-propagation-preemption` will preempt `nginx-propagation` and the deployment will be scheduled to `member3`.
```

Comment:
Do you mean that when a `PropagationPolicy` is set to `preemption: Always`, it will preempt another lower priority `PropagationPolicy` instead of **being preempted** by a higher priority `PropagationPolicy`?

Reply:
If I am an application developer, I distribute my application to a member cluster using a PropagationPolicy(`preemption: Never`). If another user preemptively uses a PropagationPolicy(`preemption: Always`) for my application, I may not even be aware of this behavior.

Reply:
I thought that when the `PropagationPolicy `is set to `preemption: Never`, it is not allowed to **be preempted**, while when the `PropagationPolicy` is set to `preemption: Always`, it is always allowed to **be preempted** by other users.

Reply:
> If I am an application developer, I distribute my application to a member cluster using a PropagationPolicy(preemption: Never). If another user preemptively uses a PropagationPolicy(preemption: Always) for my application, I may not even be aware of this behavior.

This is exactly what I'm concerned about.  I have been thinking do we need something like `I can preempt others, but not allows to be preempted`? 

Now, I think the concern could be solved by managing `Priority` properly, such as by managing who has permission to create which priority policies.

Reply:
> If I am an application developer, I distribute my application to a member cluster using a PropagationPolicy(`preemption: Never`). If another user preemptively uses a PropagationPolicy(`preemption: Always`) for my application, I may not even be aware of this behavior.

Regardless of whether this field is placed in the policy of preempting others or the policy of being preempted, there are essentially similar problems. If the above scenario is followed, then every time I want to deploy a higher priority policy and preempt, I need to perceive the old bound policy and update it(preemption: never -> always). **As an application developer, it's not convenient. It's not safe either specially when these two policies are owned by different users. You tampered with other people's policy.**

I agree with @RainbowMango. The concern could be solved by managing `Priority` properly. All preemptive behavior is defined by a set of published rules. Users need to understand the impact of the policy priority it sets on the system when it enables preemption.